adp-helm-library/templates/_helpers.tpl

```diff
@@ -20,6 +20,65 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
 {{- end -}}
 
+{{/*
+CPU and Memory requests and limits Tiers
+*/}}
+{{- define "adp-helm-library.mem-cpu-tiers" -}}
+{{- $requiredMsg := include "adp-helm-library.default-check-required-msg" . -}}
+{{- $memCpuTier := $.Values.container.memCpuTier | default "M" }}
+
+{{- $requestsMemory := "" }}
+{{- $requestsCpu := "" }}
+{{- $limitsMemory := "" }}
+{{- $limitsCpu := "" }}
+
+{{- if eq $memCpuTier "S" }}
+{{- $requestsMemory = "50Mi" }}
+{{- $requestsCpu = "50m" }}
+{{- $limitsMemory = "50Mi" }}
+{{- $limitsCpu = "50m" }}
+
+{{- else if eq $memCpuTier "M" }}
+{{- $requestsMemory = "100Mi" }}
+{{- $requestsCpu = "100m" }}
+{{- $limitsMemory = "100Mi" }}
+{{- $limitsCpu = "100m" }}
+
+{{- else if eq $memCpuTier "L" }}
+{{- $requestsMemory = "150Mi" }}
+{{- $requestsCpu = "150m" }}
+{{- $limitsMemory = "150Mi" }}
+{{- $limitsCpu = "150m" }}
+
+{{- else if eq $memCpuTier "XL" }}
+{{- $requestsMemory = "200Mi" }}
+{{- $requestsCpu = "200m" }}
+{{- $limitsMemory = "200Mi" }}
+{{- $limitsCpu = "200m" }}
+
+{{- else if eq $memCpuTier "XXL" }}
+{{- $requestsMemory = "300Mi" }}
+{{- $requestsCpu = "300m" }}
+{{- $limitsMemory = "600Mi" }}
+{{- $limitsCpu = "600m" }}
+
+{{- else if eq $memCpuTier "CUSTOM" }}
```

Comment:
@kbabigumira should we have max and min validation

Reply:
@rajeshkaremane - Discussed with Ken and we are going to use ResourceQuotas per team namespace to set a limit on how much memory and cpu a team is allowed.